// material-table/dist/m-table-pagination.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";

/* eslint-disable no-unused-vars */
import * as React from 'react';
import { Icon, IconButton, TableFooter, TableRow, TablePagination, Typography, withStyles } from '@material-ui/core';
/* eslint-enable no-unused-vars */

class MTablePaginationInner extends React.Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "handleFirstPageButtonClick", event => {
      this.props.onChangePage(event, 0);
    });

    _defineProperty(this, "handleBackButtonClick", event => {
      this.props.onChangePage(event, this.props.page - 1);
    });

    _defineProperty(this, "handleNextButtonClick", event => {
      this.props.onChangePage(event, this.props.page + 1);
    });

    _defineProperty(this, "handleLastPageButtonClick", event => {
      this.props.onChangePage(event, Math.max(0, Math.ceil(this.props.count / this.props.rowsPerPage) - 1));
    });
  }

  render() {
    const {
      classes,
      count,
      page,
      rowsPerPage
    } = this.props;
    return React.createElement("div", {
      className: classes.root
    }, React.createElement(IconButton, {
      onClick: this.handleFirstPageButtonClick,
      disabled: page === 0,
      "aria-label": "First Page"
    }, React.createElement(Icon, null, "first_page")), React.createElement(IconButton, {
      onClick: this.handleBackButtonClick,
      disabled: page === 0,
      "aria-label": "Previous Page"
    }, React.createElement(Icon, null, "chevron_left")), React.createElement(IconButton, {
      onClick: this.handleNextButtonClick,
      disabled: page >= Math.ceil(count / rowsPerPage) - 1,
      "aria-label": "Next Page"
    }, React.createElement(Icon, null, "chevron_right")), React.createElement(IconButton, {
      onClick: this.handleLastPageButtonClick,
      disabled: page >= Math.ceil(count / rowsPerPage) - 1,
      "aria-label": "Last Page"
    }, React.createElement(Icon, null, "last_page")));
  }

}

const actionsStyles = theme => ({
  root: {
    flexShrink: 0,
    color: theme.palette.text.secondary,
    marginLeft: theme.spacing.unit * 2.5
  }
});

const MTablePagination = withStyles(actionsStyles, {
  withTheme: true
})(MTablePaginationInner);
export default MTablePagination;
